Source/UI/NewUI/PaymentForm/View/ExpiryDateView.swift

```diff
@@ -41,30 +41,33 @@ public final class ExpiryDateView: UIView {
     dateInputView.update(style: self.style)
   }
 
-  func updateExpiryDate(to newDate: String?) {
-    guard newDate?.count == dateFormatTextCount else {
-      updateErrorViewStyle(isHidden: false, textfieldText: newDate, error: .invalidYear)
-      return
+  func updateExpiryDate(to textfieldText: String, replacementText: String) -> Bool {
```

Comment:
Does a method `updateExpiryDate` need to know about a textField and a replacement text? Looks like the only change from previous logic is to append something to the end of a string, which was already done at calling site. Why do you want to move it here, is there any improvement in the syntax or flow?

Reply:
Yes, it needs to know both.

Reply:
You went from a calcite syntax of:
```swift
let dateStringInput = "01/01"
dateView.updateExpiryDate(to: dateStringInput)
```

to

```swift
let textFieldText = "01/0"
let replacementText = "1"
let isValidDate = dateView.updateExpiryDate(to: textFieldText, replacementText: replacementText)
```

Which API would you think to be clearer at use site for code legibility? For reference, I'm not saying right or wrong, I'm only trying to compare readability for consumer of this method, as I would like code to read cleanly and allow a good level of confidence into what's going on without digging into each path all the way.

Reply:
Renamed